### PR TITLE
Added PathMakePath function.

### PR DESCRIPTION
**Pull request by akallabeth**
* Added ```PathMakePath``` function to allow creating directories recursively.
* Added Unit test for proper operation of function.
* Replaced ```CreateDirectoryA``` occurrences with  ```PathMakePath``` where appropriate.

### CODE REVIEW

**Comment by mfleisz** [comment]
This should fix https://github.com/FreeRDP/FreeRDP/issues/2674

File: winpr/libwinpr/path/shell.c [hunk 1]
```diff
@@ -420,6 +421,50 @@ char* GetCombinedPath(const char* basePath, const char* subPath)
...
+		if (!PathFileExistsA(copy))
+		{
+			if (!CreateDirectoryA(copy, NULL))
+				break;
+		}
+		if (strlen(copy) < length)
+			copy[strlen(copy)] = delim;
+		else
+			break;
+	}
```
Comment (hardening) [inline winpr/libwinpr/path/shell.c]: Here I would 
if (strlen(copy) >= length)
  break;
copy[strlen(copy)] = delim;

**Comment by mfleisz** [comment]
+1

**Comment by llyzs** [comment]
+1

**Pull request merged**